fpm/src/fpm.f90

```diff
@@ -85,11 +88,20 @@ subroutine build_model(model, settings, package, error)
         end if
 
     end if
-
-    call resolve_module_dependencies(model%sources)
+    if(settings%list)then
+         do i=1,size(model%sources)
+            write(stderr,'(*(g0,1x))')'fpm::build<INFO>:file expected at',model%sources(i)%file_name, &
+             & merge('exists        ','does not exist',exists(model%sources(i)%file_name) )
+         enddo
+         stop
+    else
+       call resolve_module_dependencies(model%sources)
+    endif
 
 end subroutine build_model
-
+!===================================================================================================================================
+!()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()!
+!===================================================================================================================================
```

Comment:
Those visual separators don't add much information, I would recommend to not introduce them to fpm.

First, 133 characters will also most certainly wrap on many terminal windows and they will wrap really ugly on small terminals.
Second, tools like ford use the first character in every comment to determine its purpose, this one will be fine, but ford is generally very fragile handling visual separators like this in my experience.
Finally, I think having a visual separator in a file is a good sign to split the module into several modules, I would prefer to have small self-explaining modules than long source files.